src/Controller/WidgetController.php

```diff
@@ -66,10 +68,14 @@ public function receipts(): Response
      * @param Request $request
      * @return Response|null
      */
-    public function applicationGraph(Request $request): ?Response
+    public function applicationGraph(Request $request, Department $department, Semester $semester): ?Response
     {
-        $department = $this->getDepartmentOrThrow404($request);
-        $semester = $this->getSemesterOrThrow404($request);
+        if (is_null($department)) {
+            $department = $this->getDepartmentOrThrow404($request);
+        }
+        if (is_null($semester)) {
+            $semester = $this->getSemesterOrThrow404($request);
+        }
```

Comment:
The function `getCurrentSemester` will create a new semester if there is none. Use that function instead.
![image](https://user-images.githubusercontent.com/46197518/198672450-f1676315-cc71-4d36-b5ec-8ca7b3c786b8.png)